hw/ip/entropy_src/doc/_index.md

```diff
@@ -10,83 +10,143 @@ This module conforms to the [Comportable guideline for peripheral functionality.
-Likewise, no conditioning function will be performed in this block.
-For the generation of entropy, this block provides the entropy bits needed for the entropy source solution as required by the RNG mechanism.
+This IP block provides an entropy source that is capable of using a physical noise source to generate random values in a manner that is compliant both with FIPS (though NIST SP 800-90B) and CC (AIS31) recommendations.
+
+<!-- URLS for the following pages not upstream yet, but coming -->
+The random values generated by this block serve as non-deterministic seeds for the CSRNG block.
+The outputs of the CSRNG are then used either directly by firmware or are distributed to other hardware blocks through the Entropy Distribution Network.
+
+In the terms of AIS31 classes, this block is meant to satisfy the requirements for a PTG.2 class physical entropy source, with "internal" entropy (an AIS31 term, meaning the min-entropy as measured just *before the output pins*) exceeding 0.997 entropy-bits/output-bit.
+In NIST terms, this block satisfies the requirements for a "full-entropy" source, which requires even smaller deviations from ideal entropy, at the level of less than one part in 2<sup>64</sup>.
+When operating in full-compliance mode, raw outputs from a physical digital noise source (external to this block) are passed through a conditioning function, a secure hash function (SHA-2 or SHA-3) which has been vetted by NIST to meet these stringent requirements.
+
+These raw noise bits are subjected to a sequence of health-checks to screen the raw signals for statistical defects which would cause any significant deviations from ideal entropy at the output of the conditioning block.
+These tests include:
+- The Repetition Count test, which screens for stuck-bits, or a complete failure of the physical noise source,
+- The Adaptive Proportion test, which screens for statistical bias in the number of 1's or 0's output by the noise source,
+- The "Bucket Test", which looks for correlations between the individual noise channels that the external noise source concatenates together to produce the raw noise sequence,
+- The "Markov Test", which looks for unexpected first-order temporal correlations between bits output by the individual noise channels,
+- The "Mailbox Test", in which raw-noise data is transferred to firmware in contiguous 1k-blocks, so that firmware can perform custom tests, and signal a failure through the same path as the other tests, and
+- Optional Vendor Specific tests, which allow silicon creators to extend the health checks by adding a top-level block external to this IP.
+
+The Repetition Count and Adaptive Proportion test are specifically recommended by SP 800-90B, and are implemented in accordance with those recommendations.
+In FIPS/CC-compliance mode, these latter three checks are performed on fixed window of data, typically consisting of 1024 bits each.
```

Comment:
Offline Comment from @zi-v: "Why latter three? Doesn't RCT run on 1024-bit as well?"

Reply:
Hi @zi-v, Please take a look at the expanded explanation in the text here.  The pseudo-code in SP 800-90B is pretty clear that it doesn't take a window, it's just an infinite stream.

Reply:
Right. 
I am imagining a usage where we only run 1024 bits and then shut off the RNG until next usage, but this may not necessarily be the chosen configuration.

Reply:
I see.   In that case it would run for the 1024 bits and then pause.  I guess it would need to reset it's counters when every thing is disabled just for good measure.   I'll make this explicit in the next commit.